File: oim-jruby/create_group.rb
require 'java'
require 'xlclient'

include_class('java.lang.Exception') {|package,name| "J#{name}" }
include_class 'java.lang.System' 
include_class 'java.util.HashMap'
include_class('Thor.API.tcUtilityFactory') {|package,name| "OIM#{name}"}


grpName = 'Baban'

xlclient = XLAPIClient.new
xlclient.defaultLogin

grpIntf = xlclient.getUtility('grp')

grpHash = {
  'Groups.Group Name' => grpName
}

grpMap = HashMap.new(grpHash)

t1 = System.currentTimeMillis
grpKey = grpIntf.createGroup(grpMap)
t2 = System.currentTimeMillis

delta = t2-t1
puts "Created grp with key = #{grpKey} time = #{delta}"

xlclient.close
System.exit 0

File: oim-jython/create_group.py
from xlclient import *
from group import *

import sys

if __name__ == '__main__':
    groupName = 'coders'
    xlclient = XLAPIClient('xelsysadm', 'xelsysadm')
    oimGroup = OIMGroup(xlclient)
    oimGroup.init(groupName)
    oimGroup.create()
    xlclient.close()
    sys.exit(0)
